src/Core/Configuration/AvifExtensionChecker.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Configuration;
+
+use PrestaShop\PrestaShop\Core\Image\Exception\AvifUnavailableException;
+
+/**
+ * Class AvifExtensionChecker provides object-oriented way to check if AVIF extension is installed and available.
+ */
+class AvifExtensionChecker
+{
+    private $psAdditionalImageAvif;
+
+    public function __construct($psAdditionalImageAvif)
+    {
+        $this->psAdditionalImageAvif = $psAdditionalImageAvif;
+    }
+
+    public function isAvailable()
+    {
+        $isAvailable = extension_loaded('gd') && /* @phpstan-ignore-line */
+            version_compare(PHP_VERSION, '8.1') >= 0 &&
```

Comment:
I just thought of something, this condition will always return false if `psAdditionalImageAvif` is false

What if someone configures his server to make it compatible with avif, how can he set `PS_ADDITIONAL_IMAGE_AVIF` to true if the checkbox won't be displayed since this method will always return false ?

or am I missing something ?